### PR TITLE
cat: adds --separator option to cat command

### LINKED ISSUE #6968: feat: Add option to `rclone cat` that injects a separator between files
#### The associated forum post URL from `https://forum.rclone.org`

https://forum.rclone.org/t/print-new-line-between-files-when-using-rclone-cat/37732/8

#### What is your current rclone version (output from `rclone version`)?

```
❯ rclone version
rclone v1.62.2
- os/version: ubuntu 20.04 (64 bit)
- os/kernel: 5.15.90.1-microsoft-standard-WSL2 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.20.2
- go/linking: static
- go/tags: none
```

#### What problem are you are trying to solve?

I am trying to output several remote files from an S3 bucket to stdout. Each file contains a single line. I'd like each file to output to a separate line also. However, rclone cat joins the files without any separator, which makes the output unusable/unreadable. Is there any way to accomplish this?

#### How do you think rclone should be changed to solve that?

The simplest idea I had is to add an option to `rclone cat` that will print a "separator" string after outputting each file. E.g.

```
rclone cat s3:bucket/path --include */file.txt --separator '\n'
file1
file2
file3
...
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

### PR DESCRIPTION
**Pull request by lorengordon**
#### What is the purpose of this change?

When using `rclone cat` to print the contents of several files, the
user may want to inject some separator between the files, such as a
comma or a newline. This patch adds a `--separator` option to the `cat`
command to make that possible. The default value remains an empty
string, `""`, maintaining the prior behavior of `rclone cat`.

Closes #6968

#### Was the change discussed in an issue or in the forum before?

* https://forum.rclone.org/t/print-new-line-between-files-when-using-rclone-cat/37732/14
* https://github.com/rclone/rclone/issues/6968

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: fs/operations/operations.go [hunk 1]
```diff
@@ -1301,6 +1301,14 @@ func Cat(ctx context.Context, f fs.Fs, w io.Writer, offset, count int64) error {
 			err = fs.CountError(err)
 			fs.Errorf(o, "Failed to send to output: %v", err)
 		}
+		if len(sep) >= 0 {
+			sepReader := bytes.NewReader(sep)
```
Comment (ncw) [inline fs/operations/operations.go:1305]: This will do to write the sep to the stream, no need to make a reader.

   _ , err =  w.Write(sep)

**Review by ncw — COMMENTED** [review]
Looking good :-) I put a comment inline and I've given the tests another whirl.

File: fs/operations/operations.go (hunk 1 above)
Comment (lorengordon) [reply fs/operations/operations.go]: Ok, I was wondering about something like that, but wasn't sure if there was a reason to use `io.Copy()`. Updated, ran the test, and confirmed the behavior for my use case also!

**Review by ncw — APPROVED** [review]
This looks perfect - thank you :-)

I'll merge it now.

**Pull request merged**